domain/domain.go

```diff
@@ -188,21 +188,24 @@ func (do *Domain) loadInfoSchema(startTS uint64) (infoschema.InfoSchema, bool, i
 	// 1. Not first time bootstrap loading, which needs a full load.
 	// 2. It is newer than the current one, so it will be "the current one" after this function call.
 	// 3. There are less 100 diffs.
+	// 4. No regenrated schema diff.
 	startTime := time.Now()
 	if currentSchemaVersion != 0 && neededSchemaVersion > currentSchemaVersion && neededSchemaVersion-currentSchemaVersion < 100 {
-		is, relatedChanges, err := do.tryLoadSchemaDiffs(m, currentSchemaVersion, neededSchemaVersion)
-		if err == nil {
-			do.infoCache.Insert(is, startTS)
-			logutil.BgLogger().Info("diff load InfoSchema success",
-				zap.Int64("currentSchemaVersion", currentSchemaVersion),
-				zap.Int64("neededSchemaVersion", neededSchemaVersion),
-				zap.Duration("start time", time.Since(startTime)),
-				zap.Int64s("phyTblIDs", relatedChanges.PhyTblIDS),
-				zap.Uint64s("actionTypes", relatedChanges.ActionTypes))
-			return is, false, currentSchemaVersion, relatedChanges, nil
+		is, relatedChanges, regenerateSchemaMap, err := do.tryLoadSchemaDiffs(m, currentSchemaVersion, neededSchemaVersion)
```

Comment:
Can we just return an error and log it? I think that is better than adding another boolean flag.